### PR TITLE
External Wallet Component - ViewModel

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [ ] Chore
- [X] Feature
- [ ] Bug fix

### CODE REVIEW

File: SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/components/wallets/compose/ExternalWalletsView_Loading.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        // -- Vars
+        val (loader) = createRefs()
+
+        // -- Content
+        Component.CreateLoader(modifier =
+            Modifier.constrainAs(loader) {
+                centerHorizontallyTo(parent)
+                centerVerticallyTo(parent)
+            },
+            message = "Loading your wallets"
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/components/wallets/compose/ExternalWalletsView_Loading.kt]: Localize

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/components/wallets/view/ExternalWalletViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        Assert.assertEquals(externalWalletViewModel.addressScannedValue.value, code)
+    }
+
+    @Test
+    fun test_handleQRScanned_With_Dots() {
+
+        // -- Given
+        val externalWalletViewModel = ExternalWalletViewModel()
+        val codeOne = "bitcoin:123456"
+        val codeTwo = "bitcoin:98765&tag=234"
```
Comment (brent-cybrid) [inline SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/components/wallets/view/ExternalWalletViewModelTest.kt:220]: What happens if there are more parts? Does the parsing work?

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/mock/ExternalWalletsApiMock.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            guid = "1234",
+            name = "Test",
+            asset = "BTC",
+            environment = ExternalWalletBankModel.Environment.sandbox,
+            bankGuid = "1234",
+            customerGuid = "1234",
+            address = "0x1234",
+            tag = "1234",
+            createdAt = java.time.OffsetDateTime.now(),
+            state = ExternalWalletBankModel.State.deleted,
```
Comment (brent-cybrid) [inline SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/mock/ExternalWalletsApiMock.kt:91]: There is also the state "deleting," which is the state the wallet goes into while it is being deleted by the backend.

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/components/wallets/view/ExternalWalletViewModelTest.kt (hunk 2 above)
Comment (nalancer08) [reply SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/components/wallets/view/ExternalWalletViewModelTest.kt:220]: It works but for this we only need addess and tag!

File: SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/mock/ExternalWalletsApiMock.kt (hunk 3 above)
Comment (nalancer08) [reply SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/mock/ExternalWalletsApiMock.kt:91]: I added in the test

File: SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/components/wallets/compose/ExternalWalletsView_Loading.kt (hunk 1 above)
Comment (nalancer08) [reply SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/components/wallets/compose/ExternalWalletsView_Loading.kt]: Done!

**Review by brent-cybrid — APPROVED** [review]

**Pull request merged**